### PR TITLE
Limit of 0 disables max_bytes_per_tag_values_query

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
- Disables the `max_bytes_per_tag_values_query` limit when set to 0.
- Returns a 500 instead of 200. (Screenshot of error in Grafana below)
- Adds instance SearchTag/TagValues test. This tests the scenario where `max_bytes_per_tag_values_query` is set to 0.

image

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 * [CHANGE] metrics-generator: Changed added metric label `instance` to `__metrics_gen_instance` to reduce collisions with custom dimensions. [#1439](https://github.com/grafana/tempo/pull/1439) (@joe-elliott)
 * [FEATURE] metrics-generator: support per-tenant processor configuration [#1434](https://github.com/grafana/tempo/pull/1434) (@kvrhdn)
 * [ENHANCEMENT] Added the ability to have a per tenant max search duration. [#1421](https://github.com/grafana/tempo/pull/1421) (@joe-elliott)
-* [BUGFIX] Fix nil pointer panic when the trace by id path errors. [1441](https://github.com/grafana/tempo/pull/1441) (@joe-elliott)
+* [BUGFIX] Fix nil pointer panic when the trace by id path errors. [#1441](https://github.com/grafana/tempo/pull/1441) (@joe-elliott)
+* [BUGFIX] Don't enforce `max_bytes_per_tag_values_query` when set to 0. [#1447](https://github.com/grafana/tempo/pull/1447) (@joe-elliott)
```
Comment (annanay25) [inline CHANGELOG.md:7]: I don't really think this is a bugfix: we just have mixed behaviour throughout Tempo when it comes to limits, maybe we can move this to `CHANGE`

File: modules/ingester/instance_search_test.go [hunk 2]
```diff
@@ -145,6 +147,94 @@ func TestInstanceSearch(t *testing.T) {
 	checkEqual(t, ids, sr)
 }
 
+func TestInstanceSearchTags(t *testing.T) {
```
Comment (annanay25) [inline modules/ingester/instance_search_test.go:150]: Can we club this test with the above?

Comment (annanay25) [inline modules/ingester/instance_search_test.go:150]: Was wondering if we could write a test where we return 500 once the tag-values limit is exceeded

Comment (joe-elliott) [reply modules/ingester/instance_search_test.go:114]: added.

**Review by annanay25 — APPROVED** [review]
lgtm!

**Pull request merged**